checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub

```diff
@@ -0,0 +1,215 @@
+}
+
+class AbstractMap {
+    public boolean containsValue(@Nullable Object value);
+    public boolean containsKey(@Nullable Object key);
+    public @Nullable V get(@Nullable Object key);
+    public @Nullable V remove(@Nullable Object key);
+}
+
+class AbstractSet {
+  public boolean removeAll(Collection<?> c);
+}
+
+public interface Collection<E> extends Iterable<E> {
+    boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean containsAll(Collection<?> c);
+    boolean retainAll(Collection<?> c);
+}
+
+public interface Deque<E> extends Queue<E> {
+    boolean removeFirstOccurrence(@Nullable Object o);
+    boolean removeLastOccurrence(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean contains(@Nullable Object o);
+}
+
+public interface List<E> extends Collection<E> {
+    boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean containsAll(Collection<?> c);
+    boolean removeAll(Collection<?> c);
+    boolean retainAll(Collection<?> c);
+    int indexOf(@Nullable Object o);
+    int lastIndexOf(@Nullable Object o);
+}
+
+public interface Map<K, V> {
+    boolean containsKey(@Nullable Object key);
+    boolean containsValue(@Nullable Object value);
+    @Nullable V get(@Nullable Object key);
+    @Nullable V remove(@Nullable Object key);
+    default V getOrDefault(@Nullable Object key, V defaultValue);
+    default boolean remove(@Nullable Object key, @Nullable Object value);
+    }
+
+public interface Set<E> extends Collection<E> {
+    @Pure boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    @Pure boolean containsAll(Collection<?> c);
+    boolean retainAll(Collection<?> c);
+    boolean removeAll(Collection<?> c);
+}
+
+class SynchronousQueue {
+    @Pure public boolean containsAll(Collection<?> c);
+    public boolean removeAll(Collection<?> c);
+    public boolean retainAll(Collection<?> c);
+}
+
+public class TreeMap<K,V>
+    extends AbstractMap<K,V>
+    implements NavigableMap<K,V>, Cloneable, java.io.Serializable
+{
+    public boolean containsKey(@Nullable Object key);
+    public boolean containsValue(@Nullable Object value);
+    public @Nullable V get(@Nullable Object key);
+    public @Nullable V remove(@Nullable Object key);
+}
+
+public class TreeSet<E> extends AbstractSet<E>
+    implements NavigableSet<E>, Cloneable, java.io.Serializable
+{
+    public boolean contains(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+}
+
+class Vector {
+    public synchronized boolean containsAll(Collection<?> c);
+    public synchronized boolean removeAll(Collection<?> c);
+    public synchronized boolean retainAll(Collection<?> c);
+}
+
+package java.util.concurrent;
+
+public class ConcurrentLinkedDeque<E>
+    extends AbstractCollection<E>
+    implements Deque<E>, java.io.Serializable {
+    public boolean removeFirstOccurrence(@Nullable Object o);
```

Comment:
Both of these are documented to always reject nulls.

Reply:
The .astub file is documented as
```
// This file relaxes the specifications for the following methods:
...
removeFirstOccurrence(Object)
...
```
It sounds like you would like to change that to,
"This file relaxes the specifications for the following methods when they are documented to possibly throw NPE, but does not change the specifications when they are documented to definitely throw NPE."

Is that correct?

Reply:
Yes, that's how I was thinking of it. I think that's most the useful approach for people who want to compromise safety, since it permits all potentially safe calls but no definitely unsafe calls. Plus, I think we are currently in an intermediate state, since we don't relax signatures for other classes like `ConcurrentHashMap` when we know some of their methods always reject null.

Reply:
I agree that's better.  I will change the meaning of this pull request, as documented in that file.